src/main/kotlin/pl/edu/agh/server/domain/authentication/token/TokenRepository.kt

```diff
@@ -0,0 +1,17 @@
+package pl.edu.agh.server.domain.authentication.token
+
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
+import java.util.*
+
+interface TokenRepository : JpaRepository<Token, Long> {
+
+    @Query(
+        value = " select t from Token t" +
+            " inner join User u on t.user.id = u.id " +
+            "where u.id = :id and (t.expired = false or t.revoked = false) and t.category = 'ACCESS'",
```

Comment:
Tu nie powinno byc `t.expired = false and t.revoked = false`?

Zmiana `or` na `and`